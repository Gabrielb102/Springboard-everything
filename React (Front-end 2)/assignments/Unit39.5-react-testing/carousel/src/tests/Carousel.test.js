import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from "../Carousel";

describe("Tests for Carousel component", () => {

  it("renders", () => {
    render(<Carousel />);
  });

  it("consistently renders the right HTML", () => {
    const { asFragment } = render(<Carousel />);
    expect(asFragment).toMatchSnapshot();
  });
  
});

describe("Tests Carousel for working arrow buttons", () => {
  
  it("works when you click on the left arrow on the second image", () => {
    const { queryByTestId, queryByAltText } = render(<Carousel />);

    // move forward in the carousel just to set the stage
    const rightArrow = queryByTestId("right-arrow");
    fireEvent.click(rightArrow);
    
    // move back in the carousel
    const leftArrow = queryByTestId("left-arrow");
    fireEvent.click(leftArrow);
  
    // expect the first image to show, but not the second
    expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).toBeInTheDocument();
    expect(queryByAltText("Photo by Pratik Patel on Unsplash")).not.toBeInTheDocument();
  });

  it("works when using the left arrow to cycle to the last image from the first", () => {
    const { queryByTestId, queryByAltText } = render(<Carousel />);
    
    // move back in the carousel
    const leftArrow = queryByTestId("left-arrow");
    fireEvent.click(leftArrow);
  
    // expect the third image to show, but not the first
    expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).not.toBeInTheDocument();
    expect(queryByAltText("Photo by Josh Post on Unsplash")).toBeInTheDocument();
  });

  it("works when you click on the right arrow", () => {
    const { queryByTestId, queryByAltText } = render(<Carousel />);
  
    // expect the first image to show, but not the second
    expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).toBeInTheDocument();
    expect(queryByAltText("Photo by Pratik Patel on Unsplash")).not.toBeInTheDocument();
  
    // move forward in the carousel
    const rightArrow = queryByTestId("right-arrow");
    fireEvent.click(rightArrow);
  
    // expect the second image to show, but not the first
    expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).not.toBeInTheDocument();
    expect(queryByAltText("Photo by Pratik Patel on Unsplash")).toBeInTheDocument();
  });

})
