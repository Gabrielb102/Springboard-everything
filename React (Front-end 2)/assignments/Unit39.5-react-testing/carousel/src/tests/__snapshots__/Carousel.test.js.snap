// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Carousel component consistently renders the right HTML 1`] = `[Function]`;
